custom_components/toyota/sensor.py

```diff
@@ -208,8 +214,8 @@ def extra_state_attributes(self):
             "Remaining_time": self.coordinator.data[self.index].energy.chargeinfo.get(
                 "remaining_time", None
             ),
-            "Remaining_amount": self.coordinator.data[self.index].energy.chargeinfo.get(
-                "remaining_amount", None
+            "Status": self.coordinator.data[self.index].energy.chargeinfo.get(

```

Comment:
Why change this?

Reply:
I thought it would be more helpful to have the remaining capacity displayed directly instead of the status. This way we can also use the appropriate battery device_class.

Maybe we should think about separating out some of the extra_state_attributes and making them available as separate sensors anyway?
According to the Home Assistant Developer documentation on sensors, extra_state_attributes should be avoided if possible:

> Instead of adding extra_state_attributes for a sensor entity, create an additional sensor entity. Attributes that do not change are only saved in the database once. If extra_state_attributes and the sensor value both frequently change, this can quickly increase the size of the database.

https://developers.home-assistant.io/docs/core/entity/sensor/#properties

Reply:
The refactored version of the mytoyota library (version 0.8.0+) simplifies this in the way you describes. I think we should refactor this integration to make use of that instead 😃 

And I agree with you on separating attributes into sensors instead

Reply:
See my progress here, please feel free to give it a go https://github.com/DurgNomis-drol/ha_toyota/tree/v2

I my car only supports the most basic features, so I can't test much, that was the main reason I didn't get any further with it

Reply:
Sorry for the late reply, been a busy week. If you are up for the task, i would prefer to move to the new version (I will be helping you with it) If not, we can make a workaround now. 

Side note, some of the features this is offering right now, will probably be supported directly through the HA android app as it is now supporting Android Auto.